sources/api/schnauzer/src/helpers.rs

```diff
@@ -422,6 +400,163 @@ pub fn default(
     Ok(())
 }
 
+/// Gets the value at `idx` and unwraps it. Assumes you have already checked that `idx` is in range
+/// and says so in an error message if the value cannot be obtained.
+fn get_param<'a>(helper: &'a Helper<'_, '_>, idx: usize) -> Result<&'a Value, RenderError> {
+    Ok(helper
+        .param(idx)
+        .map(|v| v.value())
+        .context(error::Internal {
+            msg: "Missing param after confirming there are enough",
+        })?)
+}
+
+/// Get the template name if there is one, otherwise return "dynamic template"
+fn template_name<'a>(renderctx: &'a mut RenderContext<'_, '_>) -> &'a str {
+    match renderctx.get_root_template_name() {
+        Some(s) => s.as_str(),
+        None => "dynamic template",
+    }
+}
+
+/// The `ecr` helper is used to map an aws-region to the correct ECR account.
+///
+/// Initially we held all of our ECR repos in a single account, but with some
+/// regions this was no longer possible. Because the ECR repo URL includes the
+/// the account number, we created this helper to lookup the correct account
+/// number for a given region.
+///
+/// This helper takes a single initial parameter, "account" or "region". The
+/// second parameter is the region.
+///
+/// # Fallback
+///
+/// A map of region to ECR account ID is maintained herein. But if we do not
+/// have the region in our map, a fallback region and account number are
+/// returned. This would allow a version of Bottlerocket to run in a new region
+/// before this map has been updated, for example.
+///
+/// # Example
+///
+/// In this example the account number for the region will be returned.
+/// `{{ ecr "account" settings.aws.region }}`
+///
+/// In this example the region will either be returned, or if it happens to be
+/// a region for which we don't have an ECR repo, the fallback region will be
+/// returned.
+/// `{{ ecr "region" settings.aws.region }}`
+pub fn ecr(
+    helper: &Helper<'_, '_>,
+    _: &Handlebars,
+    _: &Context,
+    renderctx: &mut RenderContext<'_, '_>,
+    out: &mut dyn Output,
+) -> Result<(), RenderError> {
+    trace!("Starting ecr helper");
+    let template_name = template_name(renderctx);
+    if helper.params().len() != 2 {
+        return Err(RenderError::from(
+            error::TemplateHelperError::IncorrectNumberOfParams {
+                expected: 4,
```

Comment:
```suggestion
                expected: 2,
```
Maybe we need a proc macro that gets the maximum get_param number requested and checks it at the start of the function :) :black_joker: